tests/testthat/test_civis_ml_utils.R

```diff
@@ -22,6 +22,66 @@ str_detect_multiple <- function(string, pattern) {
          string = string, pattern = pattern)
 }
 
+test_that("get_template_ids_all_versions", {
+
+  fake_template_alias_objects <- list(list(id = 11,
+                                           objectId = 12367,
+                                           objectType = "template_script",
+                                           alias = "civis-ShapefileExport",
+                                           userId = 3001,
+                                           displayName = "Export Shapefile"),
+                                      list(id = 14,
+                                           objectId = 11219,
+                                           objectType = "template_script",
+                                           alias = "civis-civisml-training",
+                                           userId = 400,
+                                           displayName = "Model Training"),
+                                      list(id = 21,
+                                           objectId = 10615,
+                                           objectType = "template_script",
+                                           alias = "civis-civisml-training-dev",
+                                           userId = 400,
+                                           displayName = "Model Training - DEV ONLY"),
+                                      list(id = 26,
+                                           objectId = 11221,
+                                           objectType = "template_script",
+                                           alias = "civis-civisml-registration-v2-2",
+                                           userId = 1750,
+                                           displayName = "Trained Model Registration, v2.2"))
+
+  with_mock(
+     `civis::fetch_until` = function(...) fake_template_alias_objects,
```

Comment:
@patr1ckm I mocked out `fetch_until` here b/c `template_alias_objects` in the code is returned by `fetch_until`: 
```
  template_alias_objects <- fetch_until(aliases_list,
                                       limit=1000,
                                       object_type='template_script',
                                       function(x) length(x) == 0)
```
When I tried mocking out   out `aliases_list` with the line per our discussion on line yesterday

`civis::aliases_list = function(...) fake_template_alias_objects` 

produced

```
civis_ml_utils.R:39: error: get_template_ids_all_versions
argument is of length zero
Backtrace:
 1. civis::get_template_ids_all_versions()
 2. civis::fetch_until(...) R/civis_ml_utils.R:39:2
 3. civis::next_page(res) R/pagination_helpers.R:82:4
```

Am I okay to mock out `fetch_until` instead of `aliases_list`?

Reply:
Yep!